src/town_gui.cpp

```diff
@@ -71,56 +68,69 @@ struct TownAuthorityWindow : Window {
 private:
 	Town *town;    ///< Town being displayed.
 	int sel_index; ///< Currently selected town action, \c 0 to \c TACT_COUNT-1, \c -1 means no action selected.
-	Scrollbar *vscroll;
 	uint displayed_actions_on_previous_painting; ///< Actions that were available on the previous call to OnPaint()
+	TownActions enabled_actions; ///< Actions that are enabled in settings.
+	TownActions available_actions; ///< Actions that are available to execute for the current company.
 
 	/**
 	 * Get the position of the Nth set bit.
 	 *
 	 * If there is no Nth bit set return -1
 	 *
-	 * @param bits The value to search in
 	 * @param n The Nth set bit from which we want to know the position
-	 * @return The position of the Nth set bit
+	 * @return The position of the Nth set bit, or -1 if no Nth bit set.
 	 */
-	static int GetNthSetBit(uint32 bits, int n)
+	int GetNthSetBit(int n)
 	{
 		if (n >= 0) {
-			for (uint i : SetBitIterator(bits)) {
+			for (uint i : SetBitIterator(this->enabled_actions)) {
 				n--;
 				if (n < 0) return i;
 			}
 		}
 		return -1;
 	}
 
+	/**
+	 * Gets all town authority actions enabled in settings.
+	 *
+	 * @return Bitmask of actions enabled in the settings.
+	 */
+	static TownActions GetEnabledActions()
+	{
+		TownActions enabled = TACT_ALL;
+
+		if (!_settings_game.economy.fund_roads) ClrBit(enabled, 3);
+		if (!_settings_game.economy.fund_buildings) ClrBit(enabled, 5);
+		if (!_settings_game.economy.exclusive_rights) ClrBit(enabled, 6);
+		if (!_settings_game.economy.bribe) ClrBit(enabled, 7);
```

Comment:
```suggestion
		if (!_settings_game.economy.fund_roads) CLRBITS(enabled, TACT_ROAD_REBUILD);
		if (!_settings_game.economy.fund_buildings) CLRBITS(enabled, TACT_FUND_BUILDINGS);
		if (!_settings_game.economy.exclusive_rights) CLRBITS(enabled, TACT_BUY_RIGHTS);
		if (!_settings_game.economy.bribe) CLRBITS(enabled, TACT_BRIBE);
```
To avoid magic values.